### ISSUE TITLE
Updating object creates new object ID

### ISSUE THREAD

**Issue by plv2002** [ParsePlatform/Parse-SDK-iOS-OSX#379]
I am new to Parse, and I am using Parse 1.8.4 I am trying to update a object already saved on the parse backend. I use the getObjectInBackgroundWithID and supply the object ID. When I try and update with new data the app tells me it successfully save a new record and when i look on the backend, I now have two records with different object ID's. Below is my code in the update button. In my app it passes in the object ID, but I have tried hard coding the object ID with the same result.  I did a search and it suggested removing the "else if let accountInfo = accountInfo {" to just "else".  After doing that i get the error "Cannot subscript a value of type 'PFObject?' when I try and assign values to the accountInfo["KeyName"]. 

@IBAction func updateAccountPressed(sender: AnyObject) {
        let query = PFQuery(className:"Accounts")

        query.getObjectInBackgroundWithId("dS7fHCoabI") {
            (accountInfo: PFObject?, error: NSError?) -> Void in
            if error != nil {
                print(error)
            } else if let accountInfo = accountInfo {
               
                accountInfo["Type"] = self.pickedAccontType
                accountInfo["NumberOrNickname"] = self.accountNumNickname.text
                accountInfo["InitialBalance"] = self.initialBalance.text
                accountInfo.saveInBackground()
            }
        }
    }

**Comment by nlutsenko** [comment]
Hey @plv2002, this sounds very strange...
The code above looks absolutely valid to me and it should simply update your accountInfo object with the new data, since you have an `objectId` on it in the returned result.

Few things here to make sure we cover all bases:
- Can you try printing out the `objectId` of `accountInfo` in the `if let ... = ...` scope. It should match the previous `objectId`
- How do you see the feedback that the app saved the object? `saveInBackground()` call is fully asynchronous and it's not guaranteed to finish saving instantly or on the same thread.

Basically - I am trying to isolate the issue to this specific code snippet.
Also, if you could attach a sample project (no need for keys, just app name is good enough) that reproduces the issue - that would be the most helpful piece, since we will be able to go and help you almost instantly.

**Comment by plv2002** [comment]
Well I feel stupid.  I was double checking what you said and I noticed that the alert it was displaying was from my addAccount function.  When I checked my storyboard somehow I linked the Update button to both my Update and Add functions.  Thanks for for the feedback next time I'll double check before opening the issue.

**Comment by nlutsenko** [comment]
Happy to help!
